library/Vanilla/AddonManager.php

```diff
@@ -1216,9 +1216,6 @@ public function getCacheDir() {
      * @return AddonManager Returns `$this` for fluent calls.
      */
     public function setCacheDir($cacheDir) {
-        if ($cacheDir !== null && strpos($cacheDir, PATH_ROOT) !== 0) {
```

Comment:
Wouldn't it be better to extend this function with new parameter for better backward compatibility:
```
    public function setCacheDir($cacheDir, bool $relative = true) {
        if ($cacheDir !== null && strpos($cacheDir, PATH_ROOT) !== 0 && $relative) {
            $cacheDir = PATH_ROOT.$cacheDir;
        }
        $this->cacheDir = $cacheDir;
        return $this;
    }
```

Reply:
What I removed wasn't backward compatible in the first place

If you read
https://github.com/vanilla/vanilla/issues/6798#issuecomment-485390630

Adam says he can't think of any reason why it would be a problem. 

It defeats the whole ability to set cache root, which used to be possible. Just like you can set the uploads root. 

So we have a situation where is some of the time cache root is correct and in other case it is needlessly appending the path root. So you are in some situation it is no possible to win. 

not the first time I encounter a situation like this one time a method had two parameter reversed break backward com. 

You can't enforce backward compatibility of a change that is already breaking backward compatibility, especially for no good reason. 

The is whole architecture is fairly new, and those line where included based on a wrong assumption.

Reply:
It basically breaks many applications of multi-tenancy setups or scalable setus, where symlinks are not viable.

Reply:
Sorry, I can't `$relative = false` since it will break backward compatibility.

Reply:
well it obviously would but @charrondev implied it was better.  

```
We have more stringent requirements at this point about backwards compatibility, specifically because of backwards compatibility issues in the past. If you want to make this change, it will have to be in a backwards compatible manner.

There are few of options for this.

Make a new method with the defaults that make sense, and deprecate the old one.
Update the existing method with a less ideal default.
```

You introduced this breaking change not me.

Reply:
`$relative` is not even in the release candidate

Reply:
You probably did not get my message correctly.
That was an example of code which:
-  would respect backward compatibility (the way this method  was called before and the way was functioning before remains absolutely the same).
- and would bring new functionality and new behavior you need for some cases same time
- and that is acceptable to me to approve such PR

Reply:
My code is not calling the methods. The framework is through the constructor, which is ultimately a part of boostraping. 

1. I could do this entirely within the method. if the if I detect if the directory depth is deeper than the framework, which would work in some cases but not others. 
2. Could introduce another config value to force absolute paths, which is probably the more likely candidate. 

It is also a mixed bug, it is not as if there is a consistent result with absolute paths as it is stands. 

I'm supporting your breaking code, which I no idea why you implemented nor what the application. All I know is it wasn't done with developers in mind.

The developer program is almost dead after 10 years. Almost nobody supports addons over versions, it is just too much hassle. The drop off rate is stark. You are not going to get contributors if focus exclusively on your clients, and stuff like branding the dashboard. I'm left wondering why bother? There are very few of us who have stuck it out for this long.

Reply:
> Almost nobody supports addons over versions,

Which addons are we talking about?

 Just create an issue in related repository please.

Reply:
I'm not talking about addon developed by the core team, I;m pointing out the incentives for developing this framework are diminished. There is a drop of in interest in people developing the framework, and supporting their addons across version of the framework. This has ab impact on contributors.

Reply:
I will go for the config override. It wouldn't be the first an probably won't be the last, because of bugs of this nature.